src/Makefile-env.am

```diff
@@ -87,6 +86,18 @@ AM_COMMON_CPPFLAGS = \
 	-DCEPH_PKGLIBDIR=\"${pkglibdir}\" \
 	-DGTEST_USE_OWN_TR1_TUPLE=0
 
+if LINUX
+AM_COMMON_CPPFLAGS += \
+	-D_REENTRANT
+endif
+
+if SOLARIS
+AM_COMMON_CPPFLAGS += \
+       -D_PTHREADS \
+       -D_POSIX_C_SOURCE
+	
+endif
```

Comment:
It gets complicated: http://docs.oracle.com/cd/E19455-01/806-5257/6je9h033k/index.html

I chose the pure posix approach.  I didn't choose the mixed mode because of compile problems in the boost headers.